crates/ide/src/lib.rs

```diff
@@ -407,9 +407,10 @@ impl Analysis {
         &self,
         position: FilePosition,
         links_in_hover: bool,
+        documentation: bool,
```

Comment:
How come we aren't actually passing `HoverConfig` into the `hover` method :rofl: ?  @Veykril could you clean this up while we are here (or in a follow up?). Take a look at completion and assist for inspiration. Note that we'll want to remove all fields which are not used by `ide` directly from hover and put them elsewhere.